### PR TITLE
Ux rework home

### PR DESCRIPTION
**Pull request by sabrinaj**

### CODE REVIEW

File: vector/src/main/java/im/vector/adapters/HomeRoomAdapter.java [hunk 1]
```diff
@@ -169,13 +160,60 @@ public Room getRoom(int position) {
...
+        return mFilteredRooms.isEmpty();
+    }
+
+    /**
+     * Return the sum of notifications for all the displayed rooms
+     * @return
+     */
+    public int getBadgeCount() {
+        int badgeCount = 0;
+        for (Room room : mFilteredRooms) {
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/adapters/HomeRoomAdapter.java]: int notificationCount = room.getNotificationCount();

                            if (bingRulesManager.isRoomMentionOnly(room)) {
                                notificationCount = room.getHighlightCount();
                            }

File: vector/src/main/java/im/vector/view/HomeSectionView.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        shape.setCornerRadius(100);
+        shape.setColor(ContextCompat.getColor(getContext(), R.color.vector_white_alpha_50));
+        mBadge.setBackground(shape);
+    }
+
+    /**
+     * Update the views to reflect the new number of items
+     */
+    private void onDataUpdated() {
+        setVisibility(mHideIfEmpty && mAdapter.isEmpty() ? GONE : VISIBLE);
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/view/HomeSectionView.java]: getBadgeCount() should be called once.

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java [hunk 3]
```diff
@@ -982,14 +941,6 @@ private void markAllMessagesAsRead() {
...
                 public void onSuccess(Void info) {
-                    Fragment currentFragment = getSelectedFragment();
-
-                    // the user could have switched to another fragment
-                    if (currentFragment instanceof VectorRecentsListFragment) {
-                        ((VectorRecentsListFragment)currentFragment).refresh();
-                        stopWaitingView();
-                    }
-
                     refreshUnreadBadges();
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: Refresh the fragment.

File: vector/src/main/java/im/vector/adapters/HomeRoomAdapter.java (hunk 1 above)
Comment (sabrinaj) [inline vector/src/main/java/im/vector/adapters/HomeRoomAdapter.java]: Done

File: vector/src/main/java/im/vector/view/HomeSectionView.java (hunk 2 above)
Comment (sabrinaj) [inline vector/src/main/java/im/vector/view/HomeSectionView.java]: Done

**Pull request merged**